Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -266,39 +275,92 @@ def bottle_formula(f, args:)
       return
     end
 
-    return ofail "Formula was not installed with --build-bottle: #{f.full_name}" unless Utils::Bottles.built_as? f
-
     return ofail "Formula has no stable version: #{f.full_name}" unless f.stable
 
-    if args.no_rebuild? || !f.tap
-      rebuild = 0
+    bottle_tag, rebuild = if local_bottle_json
+      f.local_bottle_path.to_s =~ Pathname::BOTTLE_EXTNAME_RX
+      [Regexp.last_match(2).to_sym, Regexp.last_match(4).to_i]
```

Comment:
Bit of a nit: we should probably add a method for `BOTTLE_EXTNAME_RX` handling. Understanding what it's matching here requires looking at two files. If the regex ever changes then we need to change multiple files.

Reply:
Yes, fair.